elements/lisk-tree/src/merkle_tree.ts

```diff
@@ -11,21 +11,63 @@
  *
  * Removal or modification of this copyright notice is prohibited.
  */
+/* eslint-disable no-bitwise */
+/* eslint-disable @typescript-eslint/prefer-for-of */
 
+import { hash } from '@liskhq/lisk-cryptography';
 import { Proof } from './types';
+import { EMPTY_HASH, LEAF_PREFIX, BRANCH_PREFIX } from './constants';
 
 export class MerkleTree {
-	// eslint-disable-next-line
 	private _root: Buffer;
-	// eslint-disable-next-line
-	public constructor(_initValues: Buffer[]) {
-		this._root = Buffer.alloc(0);
+	private _dataLength = 0;
+	private _data: { [key: string]: Buffer } = {};
+
+	public constructor(initValues: Buffer[] = []) {
+		this._dataLength = initValues.length;
+		if (this._dataLength === 0) {
+			this._root = EMPTY_HASH;
+			return;
+		}
+		const leafHashes = new Array<Buffer>(this._dataLength);
+		for (let i = 0; i < this._dataLength; i += 1) {
+			const value = Buffer.concat(
+				[LEAF_PREFIX, initValues[i]],
+				LEAF_PREFIX.length + initValues[i].length,
+			);
+			const key = hash(value);
+			leafHashes[i] = key;
+			this._data[key.toString('binary')] = value;
+		}
+		const height = this._height;
+		let layer = leafHashes;
+		for (let i = 0; i < height - 1; i += 1) {
+			const currentLayerHashes: Buffer[] = [];
+			const pairBranches = layer.length >>> 1;
```

Comment:
Is this `layer.length >>> 1` not same as `layer.length / 2`? Then use simple version which is more readable.

Reply:
it's integer calculation of `/ 2`, but since it's float it will have different value unless we use `floor`.
But in general, it's faster and accurate to do the bit shifting, because it's integer calculation.